src/main/java/org/duckstudy/model/car/Cars.java

```diff
@@ -0,0 +1,73 @@
+package org.duckstudy.model.car;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.duckstudy.model.generator.Generator;
+import org.duckstudy.view.OutputView;
+
+public class Cars {
+
+    private static final int INITIAL_POSITION = 0;
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = List.copyOf(cars);
+    }
+
+    public Cars(String[] carNames, Generator generator) {
+        this.cars = createCars(carNames, generator);
+    }
+
+    private List<Car> createCars(String[] carNames, Generator generator) {
+        List<Car> cars = new ArrayList<>();
+        for (String carName : carNames) {
+            cars.add(new Car(carName, generator));
+        }
+        return List.copyOf(cars);
+    }
+
+    public Cars playAndGetWinners(int repetitionNum, OutputView outputView) {
+        for (int i = 0; i < repetitionNum; i++) {
+            moveAll();
+            outputView.printMessage(getPositions());
+        }
+        return calculateWinners();
+    }
+
+    private void moveAll() {
+        for (Car car : cars) {
+            car.move();
+        }
+    }
+
+    private Cars calculateWinners() {
+        int maxPosition = cars.stream()
+                .mapToInt(Car::getPosition)
+                .max()
+                .orElse(INITIAL_POSITION);
+
+        ArrayList<Car> winners = cars.stream()
+                .filter(car -> car.getPosition() == maxPosition)
+                .collect(ArrayList::new, ArrayList::add, ArrayList::addAll);
+        return new Cars(winners);
```

Comment:
```suggestion
        List<Car> winners = cars.stream()
                .filter(car -> car.getPosition() == maxPosition)
                .toList();
        return new Cars(winners);
```
여도 되지 않을까요? `ArrayList`는 추상화된 `List`와 다른 구현체에요. `추상화에 의존해야지, 구체화에 의존하면 안된다`라는 말이 왜 나왔을까요?

Reply:
앗 맞습니다,, 구체 클래스에 의존하면 코드 변경 가능성이 높아집니다,, 수정하겠습니다 ! 😭